src/audio_worklet_node.rs

```diff
@@ -188,44 +247,46 @@ fn process_audio_worklet(env: &Env, processors: &JsObject, args: ProcessorArgume
-                }
+            // @todo - get from global
+            let render_quantum_size = 128;
+            let mut js_inputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_inputs)?;
+            let mut js_outputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_outputs)?;
+            let mut js_params = processor.get_property::<JsSymbol, JsObject>(k_worklet_params)?;
+            let js_params_cache =
+                processor.get_property::<JsSymbol, JsObject>(k_worklet_params_cache)?;
 
-                // delete remaining channels, if any
-                for i in input.len() as u32..channels.get_array_length().unwrap() {
-                    channels.delete_element(i)?;
-                }
+            // Check JS input and output, and rebuild JS object if layout changed
+            if !check_same_io_layout(&js_inputs, inputs) {
+                let new_js_inputs = rebuild_io_layout(env, js_inputs, inputs, render_quantum_size);
+                // Store new layout in processor
+                processor.set_property(k_worklet_inputs, new_js_inputs)?;
+                // Override js_inputs with new reference
+                js_inputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_inputs)?;
             }
 
-            let js_outputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_outputs)?;
-
-            for (output_number, output) in outputs.iter().enumerate() {
-                let mut channels = js_outputs.get_element::<JsObject>(output_number as u32)?;
+            if !check_same_io_layout(&js_outputs, outputs) {
+                let new_js_outputs =
+                    rebuild_io_layout(env, js_outputs, outputs, render_quantum_size);
+                // Store new layout in processor
+                processor.set_property(k_worklet_outputs, new_js_outputs)?;
+                // Override js_outputs with new reference
+                js_outputs = processor.get_property::<JsSymbol, JsObject>(k_worklet_outputs)?;
+            }
 
-                for (channel_number, channel) in output.iter().enumerate() {
-                    let samples =
-                        float_buffer_to_js(env, channel.as_ptr() as *mut _, channel.len());
-                    channels.set_element(channel_number as u32, samples)?;
-                }
+            // Copy inputs into JS inputs buffers
+            for (input_number, input) in inputs.iter().enumerate() {
+                let js_input = js_inputs.get_element::<JsObject>(input_number as u32)?;
 
-                // delete remaining channels, if any
-                for i in output.len() as u32..channels.get_array_length().unwrap() {
-                    channels.delete_element(i)?;
+                for (channel_number, channel) in input.iter().enumerate() {
+                    let js_channel = js_input.get_element::<JsTypedArray>(channel_number as u32)?;
+                    let mut js_channel_value = js_channel.into_value()?;
```

Comment:
Would be good to check at some point if we don't allocate with some of these calls

Reply:
Normally except in `rebuild_io_layout` there should be no allocation, but clearly needs to be confirmed

I'm just thinking we could also have a pool of preallocated Float32Array to mitigate further this possible issue